crates/interfaces/src/p2p/headers/client.rs

```diff
@@ -20,12 +20,13 @@ pub struct HeadersRequest {
 #[async_trait]
 #[auto_impl::auto_impl(&, Arc, Box)]
 pub trait HeadersClient: Send + Sync + Debug {
-    /// Update the node's Status message.
-    ///
-    /// The updated Status message will be used during any new eth/65 handshakes.
-    fn update_status(&self, height: u64, hash: H256, td: U256);
-
     /// Sends the header request to the p2p network and returns the header response received from a
     /// peer.
     async fn get_headers(&self, request: HeadersRequest) -> PeerRequestResult<BlockHeaders>;
 }
+
+/// The status updater for updating the status of the p2p node
+pub trait StatusUpdater: Send + Sync {
+    /// Updates the status of the p2p node
+    fn update_status(&self, height: u64, hash: H256, total_difficulty: U256);
+}
```

Comment:
useful to have this as trait, so it can be used separately from the NetworkHandle